mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -110,13 +110,40 @@ public RunInfo createRun(CreateRun request) {
     return mapper.toCreateRunResponse(ojson).getRun().getInfo();
   }
 
-  /** @return  a list of all RunInfos associated with the given experiment. */
+  /**
+   * @return a list of all RunInfos associated with the given experiment.
+   */
   public List<RunInfo> listRunInfos(long experimentId) {
-    SearchRuns request = SearchRuns.newBuilder().addExperimentIds(experimentId).build();
+    List<Long> experimentIds = new ArrayList<>();
+    experimentIds.add(experimentId);
+    return searchRuns(experimentIds, null);
+  }
+
+  /**
+    * @return a list of all RunInfos that satisfy search filter.
+    */
+  public List<RunInfo> searchRuns(List<Long> experimentIds, String searchFilter) {
```

Comment:
We should say what shape searchFilter takes or where to find more docs for it, and indicate that it may be null (maybe use `@Nullable` if that's available).

Reply:
Could not get annotations to work. Handled `null` cases and added javadocs.